safe/gui/widgets/dock.py

```diff
@@ -931,14 +931,17 @@ def layer_changed(self, layer):
 
             if provenances:
                 self.set_provenance_to_project_variables(provenances)
-            if is_multi_exposure:
-                self.impact_function = (
-                    MultiExposureImpactFunction.load_from_output_metadata(
-                        keywords))
-            elif provenances:
-                self.impact_function = (
-                    ImpactFunction.load_from_output_metadata(
-                        keywords))
+            try:
+                if is_multi_exposure:
+                    self.impact_function = (
+                        MultiExposureImpactFunction.load_from_output_metadata(
+                            keywords))
+                elif provenances:
+                    self.impact_function = (
+                        ImpactFunction.load_from_output_metadata(
+                            keywords))
+            except Exception:
```

Comment:
This kind of `except Exception` is hiding us all possible bugs raised by these functions. I'm personally against this because the user and mainly us won't have any information about what is wrong in `load_from_metadata`. Is-it not possible to have more granular catch statement?

Reply:
+1

Reply:
fixed, I raised `InvalidLayerError` exception instead of `Extension` so we know which extension is  caught